web/client-api/src/main/java/io/deephaven/web/client/api/JsTable.java

```diff
@@ -490,6 +499,11 @@ public void setInternalViewport(double firstRow, double lastRow, Column[] column
         if (firstRow < 0) {
             throw new IllegalArgumentException(firstRow + " < " + 0);
         }
+        final ActiveTableBinding sub = state().getActiveBinding(this);
+        if (isStreamTable() && sub != null && sub.getSubscription() != null) {
```

Comment:
this feels very racy to me, having gotten bitten by this in the past. i think instead you want to test for `subscriptions.get(getHandle())`, and if non-null, check the status of that subscription...

...that said, isn't this still a ui issue, where the UI is calling setViewport unnecessarily?

--

i'd also try to push this into the TableViewportSubscription, which should manage the "is this right one?" question, and just let you say "in this case, never forward the viewport change". If i'm reading this right, you are going to end up passing through there anyway.